include/boost/astronomy/io/hdu.hpp

```diff
@@ -171,26 +112,36 @@ struct hdu
         }
 
         //setting naxis values
-        naxis_.emplace_back(cards[key_index["NAXIS"]].value<std::size_t>());
-        naxis_.reserve(naxis_[0]);
+        std::size_t total_dimensions = cards[key_index["NAXIS"]].value<std::size_t>();
```

Comment:
In old code total dimensions was also stored ( NAXIS ) which is actually not required because the size of **Naxis** vector is itself the total number of dimensions